calico/felix/fetcd.py

```diff
@@ -334,6 +335,56 @@ def __init__(self, config, etcd_api, status_reporter, hosts_ipset):
         self.last_rate_log_time = monotonic_time()
         # Register for events when values change.
         self._register_paths()
+        self._usage_count_greenlet = gevent.Greenlet(
+            self._periodically_usage_count
+        )
+
+    def estimated_host_count(self):
+        """
+        Single operations to python's built-in datastructures are thread-safe.
+        You can update a map, or add something to a set or update an integer,
+        and it's atomic.
+
+        :return An aproximation to our Felix cluster count
+        """
+        return len(self.ipv4_by_hostname)
+
+    @logging_exceptions
+    def _periodically_usage_count(self):
+        """
+        Greenlet: periodically report the cluster existence to
+        projectcalico.org.  Period is about once per day.
+
+        :return: Does not return, unless UsageCount disabled.
+        """
+        interval = 72000   # Once every 20 hours plus/minus jitter
+        if not self._config.USAGE_COUNT:
+            _log.info("usage count is disabled")
+            return
+
+        _log.info("Started usage count thread.  Usage count interval: %s", interval)
+
+        while True:
+            try:
+                config = self._config
+                calico_version = str(pkg_resources.require("calico")[0].version)
+                felix_count = self.estimated_host_count()
+                cluster_type = "NA"
+                report_usage_and_get_warnings(calico_version, config.HOSTNAME, config.CLUSTER_GUID, felix_count, cluster_type)
+            except Exception:
+                # except Exception: so that we don't catch SystemExit an KeyboardInterrupt. Those should kill the app.
+
+                # Jitter by 1% of interval (about 12 minutes)
+                jitter = random.random() * 0.01 * interval
+                sleep_time = interval + jitter
+                _log.warning("Error when usage_count. "
```

Comment:
I think I previously commented that this appears to be the retry branch (i.e. failed to connect to the usage server so will retry later), and that 24 hours seemed like too long to back off.  We had previously discussed an exponential back off up to some maximum I believe.  You basically said no this isn't the retry.  But I'm pretty sure it is.  It's in the except branch.  And it logs "Error when usage_count".

Retrying relatively soon (e.g. a few minutes)  is important because you hypothesized that AWS lambda has a maximum number of request per second and will discard if we go above that rate.  In that scenario we want to retry reasonable soon, not loose the data for that day.

The exponential back off is important because the other reason it might be failing is because the host cannot access the internet, and we don't want to spam their firewall. So maybe something like a week as the max back off?